intertidal/mosaics.py

```diff
@@ -0,0 +1,193 @@
+import click
+
+from datacube.utils.aws import configure_s3_access
+from intertidal.utils import (
+    configure_logging,
+    round_date_strings,
+)
+from pathlib import Path
+import os
+
+
+def find_files(directory, pattern):
+    path = Path(directory)
+    return list(path.rglob(f"*{pattern}"))
+
+
+def _is_s3(path):
+    """
+    Determine whether output location is on S3.
+    """
+    uu = urlparse(path)
+    return uu.scheme == "s3"
+
+
+@click.command()
+@click.option(
+    "--product",
+    type=str,
+    required=True,
+    help="The name of the product being mosaiced "
+    "eg ga_s2_tidal_composites_cyear_3 or ga_s2ls_intertidal_cyear_3. ",
+)
+@click.option(
+    "--band",
+    type=str,
+    required=True,
+    help="The name of the band to be mosaiced. eg elevation or exposure ",
+)
+@click.option(
+    "--year",
+    type=str,
+    required=True,
+    help="The year of the band to be mosaiced eg 2022 ",
+)
+@click.option(
+    "--version",
+    type=str,
+    required=True,
+    help="The version number of the product to be mosaiced (e.g. " "'0-0-1').",
+)
+@click.option(
+    "--product_dir",
+    type=str,
+    default="s3://dea-public-data-dev/derivative/",
+    help="The directory/location to read the tile cogs from; supports "
+    "both local disk and S3 locations. "
+    "Defaults to 's3://dea-public-data-dev/derivative/'.",
+)
+@click.option(
+    "--output_dir",
+    type=str,
+    default="/gdata1/projects/coastal/intertidal/mosaics/",
+    help="The directory/location to output data and metadata; supports "
+    "both local disk and S3 locations. Defaults to '/gdata1/projects/coastal/intertidal/mosaics/'. "
+    "The function will add on {product}/{version}/continental_mosaics/{year}--P1Y to the provided ouput_dir ",
+)
+@click.option(
+    "--dataset_maturity",
+    type=str,
+    default="final",
+    help="Dataset maturity metadata to used in the name of the input datasets and output dataset. "
+    "Defaults to 'final', can also be 'interim'.",
+)
+def make_mosaic_cli(
+    product,
+    band,
+    year,
+    version,
+    product_dir,
+    output_dir,
+    dataset_maturity="final",
+    sunglint="",
+):
+    input_params = locals()
+    run_id = f"[{version}] [{year}] [{band}]"
+    log = configure_logging(run_id)
+
+    # Record params in logs
+    log.info(f"{run_id}: Using parameters {input_params}")
+
+    s3 = _is_s3(product_dir)
+    product_dir = product_dir.replace("s3://", "")
+    product_dir = product_dir.rstrip("/")
+    product_dir = f"{product_dir}/{product}/{version}"
+
+    log.info(f"{run_id}: Using product_dir {product_dir}")
+    output_dir = output_dir.rstrip("/")
+    output_dir = f"{output_dir}/{product}/{version}/continental_mosaics/{year}--P1Y"
+
+    run_id = f"[{product}] [{band}] [{year}]"
+    log = configure_logging(run_id)
+
+    fs = s3fs.S3FileSystem(anon=True)
+    # Configure S3
+    configure_s3_access(cloud_defaults=True, aws_unsigned="--aws_unsigned")
```

Comment:
`--aws_unsigned` is probably not a valid input; I think it needs to be True or False

Reply:
modified the use of aws_unsigned to match intertidal